grpc/src/main/java/com/linecorp/armeria/server/grpc/HttpJsonTranscodingService.java

```diff
@@ -830,4 +886,21 @@ public String toString() {
             }
         }
     }
+
+    /**
+     * Notifies that a recursively nesting type exists.
+     */
+    private static class RecursiveTypeException extends IllegalArgumentException {
+        private static final long serialVersionUID = -6764357154559606786L;
+
+        private final Descriptor recursiveTypeDescriptor;
+
+        RecursiveTypeException(Descriptor recursiveTypeDescriptor) {
+            this.recursiveTypeDescriptor = recursiveTypeDescriptor;
+        }
+
+        Descriptor recursiveTypeDescriptor() {
+            return recursiveTypeDescriptor;
+        }
```

Comment:
Could override `fillInStackTrace()` with no-op to reduce the overhead.

Reply:
👍 Good point! Added.